app/views/submissions/show.html.slim

```diff
@@ -62,6 +75,14 @@ javascript:
               = link_to '<span class="fa fa-edit"></span>'.html_safe, edit_submission_path(@submission), class: "btn btn-primary"
               br
 
+      - if logged_in? && (current_user.is_moderator || current_user.is_admin)
```

Comment:
Nitpick: I'd recommend defining a method like `current_user&.can_soft_delete?` containing this login instead of repeating this `||` check everywhere. Feel free to disregard.